orchagent/dash/dashaclorch.cpp

```diff
@@ -0,0 +1,889 @@
+            if (*(acl_rule.m_terminating))
+            {
+                attrs.back().value.s32 = SAI_DASH_ACL_RULE_ACTION_PERMIT;
+            }
+            else
+            {
+                attrs.back().value.s32 = SAI_DASH_ACL_RULE_ACTION_PERMIT_AND_CONTINUE;
+            }
+        }
+        else
+        {
+            if (*(acl_rule.m_terminating))
+            {
+                attrs.back().value.s32 = SAI_DASH_ACL_RULE_ACTION_DENY;
+            }
+            else
+            {
+                attrs.back().value.s32 = SAI_DASH_ACL_RULE_ACTION_DENY_AND_CONTINUE;
+            }
+        }
+    }
+
+    if (updateValue(data, "protocol", acl_rule.m_protocols) || (acl_rule.m_protocols && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_RULE_ATTR_PROTOCOL;
+        attrs.back().value.u8list.count = static_cast<uint32_t>(acl_rule.m_protocols.value().size());
+        attrs.back().value.u8list.list = acl_rule.m_protocols.value().data();
+    }
+
+    if (updateValue(data, "src_addr", acl_rule.m_src_ips) || (acl_rule.m_src_ips && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_RULE_ATTR_SIP;
+        attrs.back().value.ipaddrlist.count = static_cast<uint32_t>(acl_rule.m_src_ips.value().size());
+        attrs.back().value.ipaddrlist.list = acl_rule.m_src_ips.value().data();
+    }
+
+    if (updateValue(data, "dst_addr", acl_rule.m_dst_ips) || (acl_rule.m_dst_ips && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_RULE_ATTR_DIP;
+        attrs.back().value.ipaddrlist.count = static_cast<uint32_t>(acl_rule.m_dst_ips.value().size());
+        attrs.back().value.ipaddrlist.list = acl_rule.m_dst_ips.value().data();
+    }
+
+    if (updateValue(data, "src_port", acl_rule.m_src_ports) || (acl_rule.m_src_ports && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_RULE_ATTR_SRC_PORT;
+        attrs.back().value.u16rangelist.count = static_cast<uint32_t>(acl_rule.m_src_ports.value().size());
+        attrs.back().value.u16rangelist.list = acl_rule.m_src_ports.value().data();
+    }
+
+    if (updateValue(data, "dst_port", acl_rule.m_dst_ports) || (acl_rule.m_dst_ports && !is_existing))
+    {
+        attrs.emplace_back();
+        attrs.back().id = SAI_DASH_ACL_RULE_ATTR_DST_PORT;
+        attrs.back().value.u16rangelist.count = static_cast<uint32_t>(acl_rule.m_dst_ports.value().size());
+        attrs.back().value.u16rangelist.list = acl_rule.m_dst_ports.value().data();
+    }
+
+    if (!is_existing)
+    {
+        // Mandatory on create attributes should be setted, otherwise assign the default value.
+        // If the attributes don't have default value, just skip and wait for the user to set the value at the next message
+        if (!acl_rule.m_action || !acl_rule.m_terminating)
+        {
+            acl_rule.m_action == DashAclRuleEntry::Action::ALLOW;
```

Comment:
Are these default values defined somewhere/how did we decied on these?

Reply:
To the default value of protocol, just refer: https://github.com/Azure/DASH/blob/main/documentation/general/dash-sonic-hld.md#324-acl
```
if not provided, match on all protocols
```
To the action, just explicit assigning according to the SAI definition(https://github.com/opencomputeproject/SAI/blob/dash/experimental/saiexperimentaldashacl.h#L98).
But per our previous discussion, I can remove these explicit assigning.